### ISSUE TITLE
[$250] IOU - In newly created workspace, with no members invited, split expense is displayed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46739]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.16
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch app
2. Tap profile icon -- workspaces -- new workspace
3. Navigate to newly created workspace in the list and using 3 dots on beside navigate to #admin room
4. Tap plus icon near compose and select split expense.
5. Enter an amount and tap next

## Expected Result:
In newly created workspace, with no members invited, split expense option must not be displayed in #admin room. When no members invited in workspace, it must not show hidden member

## Actual Result:
In newly created workspace, with no members invited, split expense option is displayed in #admin room. When trying to split expense, a hidden member is shown without invited

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details

https://github.com/user-attachments/assets/e01f812e-aab0-4f3a-bdb3-fa434f487b75

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0125de9b07488ab7a1</li>
        <li>Upwork Job ID: 1820610996701136238</li>
        <li>Last Price Increase: 2024-08-06</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Reviewer | 103461530</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ahmedGaber93</details>

**Comment by lanitochka17** [comment]
@adelekennedy  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by rlmorrison74** [comment]
I really apologize for commenting here, but I don't know where else to go that someone from expensify might see it. I emailed contributors@expensify.com a week ago requesting access to slack, along with a follow up a few days ago, and I haven't heard back. Are we just not accepting new contributors at the moment, or is there some other step I need to follow to gain access? Can someone please advise? Again, sorry for hijacking an issue thread.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
IOU - In newly created workspace, with no members invited, split expense is displayed

### What is the root cause of that problem?
We are including split option if it your own expense chat here

https://github.com/Expensify/App/blob/8abdcca86923a5a121734f34bbc53f2eeb5a45dd/src/libs/ReportUtils.ts#L6106-L6107

### What changes do you think we should make in order to solve the problem?

Change it to 
```
        (isPolicyExpenseChat(report) && report?.isOwnPolicyExpenseChat && otherParticipants.length > 0)

```

### What alternative solutions did you explore? (Optional)

**Comment by kpadmanabhan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace with 0 members shows Split expense option.

### What is the root cause of that problem?
We do not check count of members for displaying Split expense option in context menu.

### What changes do you think we should make in order to solve the problem?
Check for members count in
`src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.tsx`

### What alternative solutions did you explore? (Optional)

**Comment by hayes102** [comment]
I am not also able to reproduce the issue ... the split option doesn't appear in the admin room

**Comment by adelekennedy** [comment]
reproduced - I'm also able to reproduce this on web, chrome v9.0.16-5, IMO this is a bug, we shouldn't show the split option until additional members are invited

**Comment by adelekennedy** [comment]
@ahmedGaber93 already some proposals to review!

**Comment by adelekennedy** [comment]
@rlmorrison74 Let me search for your email in the inbox - I'll reach out via email fro next steps

**Comment by ahmedGaber93** [comment]
I'm not able to reproduce on web, do I missed something?


https://github.com/user-attachments/assets/5ae41726-b6dd-4c14-b319-83d4e3f4dc3b

**Comment by ahmedGaber93** [comment]
> You can try the new account, with the first workspace in that account.

@daledah Still not able to reproduce


https://github.com/user-attachments/assets/b5383772-1d31-4ec7-b0c4-6dd0d6b71c41

**Comment by ahmedGaber93** [comment]
I am able to reproduce now. Thanks @daledah.

**Comment by ahmedGaber93** [comment]
@daledah's [proposal](https://github.com/Expensify/App/issues/46739#issuecomment-2266645338) LGTM!

🎀 👀 🎀 C+ reviewed

**Comment by ahmedGaber93** [comment]
Not overdue, PR in progress.

**Comment by trjExpensify** [comment]
This PR caused a pretty big regression where none of us with expensify.com emails can use the create options in DM: https://github.com/Expensify/App/issues/48282

**Comment by ahmedGaber93** [comment]
@trjExpensify Yes, unfortunately this is a big regression, but the good news is the PR is not merged into production yet. I will create a revert PR now.

**Comment by trjExpensify** [comment]
Oh hm, but it's on a checklist that's closed? 🤔 CC: @puneetlath is something broken? 

<img width="981" alt="image" src="https://github.com/user-attachments/assets/07379b19-2737-4eed-b1b2-0767b99d3a46">

**Comment by puneetlath** [comment]
We didn't do a production deploy for that checklist, so it's still on staging.

**Comment by ahmedGaber93** [comment]
@daledah I think this not urgent issue and there is no need to rush, let's revert it before deploying to production then we can discuss the solution. May be emails with `GUIDES_DOMAIN` also need to use this feature on other scopes.

**Comment by puneetlath** [comment]
I merged and am CPing the revert.

**Comment by puneetlath** [comment]
Confirmed the options appear after reverting 

<img width="465" alt="image" src="https://github.com/user-attachments/assets/d4924d16-684e-4132-a5da-35d9dc631775">

**Comment by ahmedGaber93** [comment]
@adelekennedy the expect behavior here is to hide the split option in admin room if the only other user in the room is guide team member and there is no other members. 

The question here: is there is any case the guide member with email `team.expensify.com` can be normal user of admin room and it is not a guide member? For example, if we have a workspace for guide team, the guide team members here will be normal users on admin room not guide member

CC @trjExpensify @Gonals

**Comment by adelekennedy** [comment]
I'm not sure  but my instinct is that they'd always be guide members but @trjExpensify or @Gonals for thoughts? I'll take this to Slack to confirm if we don't come to a conclusion here

**Comment by trjExpensify** [comment]
>is there is any case the guide member with email team.expensify.com can be normal user of admin room and not a guide member? For example, if we have a workspace for guide team, the guide team members here will be normal users on admin room not guide member

What do you mean by normal user? They can have their own workspaces for testing and demos on that domain, if that's the question.

**Comment by trjExpensify** [comment]
Does anyone have the latest on what we do with Guides/AMs now? Do we just add them as a member of the #admins room, versus adding them to the workspace when assigned to a lead? CC: @puneetlath 

If that's the case, then I think it's probably fine to just show the guide/AM as a member of the #admins room and thus as a participant here: 

<img width="337" alt="image" src="https://github.com/user-attachments/assets/683a199f-77b9-4a3e-a31b-8724d214f119">

**Comment by puneetlath** [comment]
> Does anyone have the latest on what we do with Guides/AMs now? Do we just add them as a member of the #admins room, versus adding them to the workspace when assigned to a lead? CC: @puneetlath

Yes, that's correct. We add them to the admins room, but not the workspace.

**Comment by trjExpensify** [comment]
Okay cool, then I think it's fine to show the guide/AM as a participant of the #admins room and not hide that. Meaning, they can be selected for a `Split` in the room, but I highly doubt anyone will do that in reality. 

So what we should try to fix here is why they're showing as hidden. Agreed?

**Comment by puneetlath** [comment]
Makes sense to me. Hiding seems to cause more problems than benefits.

**Comment by ahmedGaber93** [comment]
Thanks all for clearing the context.

**Comment by ahmedGaber93** [comment]
@adelekennedy @daledah The "hidden" participant not reproduced with me in this case, it always displays the Guide member. Is there anyone able to reproduce it?


https://github.com/user-attachments/assets/1550c22a-162e-4792-8e2a-cc515b83dd3f

**Comment by ahmedGaber93** [comment]
@daledah the issue is not reproduced on latest main with me. Can you confirm?

**Comment by ahmedGaber93** [comment]
@daledah can you try to clear the cache like you described here https://github.com/Expensify/App/issues/46739#issuecomment-2273382189 then retest.

**Comment by ahmedGaber93** [comment]
@daledah bump here https://github.com/Expensify/App/issues/46739#issuecomment-2354587449

**Comment by ahmedGaber93** [comment]
@adelekennedy The "hidden" participant is not reproduced with me in this case, it always displays the Guide member. I think we can close it unless someone able to reproduce. https://github.com/Expensify/App/issues/46739#issuecomment-2332221835

**Comment by adelekennedy** [comment]
agreed - let's close this one!

**Comment by adelekennedy** [comment]
ah good catch - sorry about that I agree payment is due here

Payouts due:

- [ ] Contributor: $250 @daledah [Upwork](https://www.upwork.com/ab/applicants/1844052997589655962/suggested)
- [ ] Contributor+: $250 @ahmedGaber93 [Upwork](https://www.upwork.com/ab/applicants/1844052997589655962/suggested)

@daledah @ahmedGaber93 will you apply [here](https://www.upwork.com/jobs/~021844052997589655962)? Upwork is letting he search talent at the moment

**Comment by adelekennedy** [comment]
@daledah sent @ahmedGaber93 just sent offer, please accept and I will pay you

**Comment by ahmedGaber93** [comment]
Hi, @adelekennedy @daledah what is left here to close this issue?

**Comment by adelekennedy** [comment]
I'm so sorry for the delay here - @daledah payment just made!

**Comment by ahmedGaber93** [comment]
I think we can close it now.
